e2e-tests/src/test/button_game/contracts.rs

```diff
@@ -0,0 +1,222 @@
+use aleph_client::{
+    contract::{
+        util::{to_account_id, to_bool, to_u128},
+        ContractInstance,
+    },
+    AnyConnection, Balance, Connection, KeyPair, SignedConnection,
+};
+use anyhow::{Context, Result};
+use sp_core::{
+    crypto::{AccountId32, AccountId32 as AccountId, Ss58Codec},
+    Pair,
+};
+
+use crate::Config;
+
+pub trait AsContractInstance {
```

Comment:
instead of all these three trait's I'd suggest implementing `impl From<Foo> for Bar`, as it is more customary and gives you both from/into for free